x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -141,6 +149,22 @@ export function insertOrReplaceColumn(args: ColumnChange): IndexPatternLayer {
   return insertNewColumn(args);
 }
 
+export function ensureCompatibleFiltersAreMoved<
+  T extends {
+    operationType: OperationType;
+    filter?: unknown;
+    timeShift?: unknown;
+  }
+>(column: T, referencedOperation: GenericOperationDefinition, previousColumn: ColumnFilters) {
+  if (referencedOperation.filterable) {
+    column.filter = (previousColumn as ReferenceBasedIndexPatternColumn).filter;
+  }
+  if (referencedOperation.shiftable) {
+    column.timeShift = (previousColumn as ReferenceBasedIndexPatternColumn).timeShift;
+  }
+  return column;
```

Comment:
This function is returning the column but also mutating it. We had issues with this kind of functions before - could we make this immutable so a new column with filter/timeshift is returned?